datatypes/src/collections/multi_point_collection.rs

```diff
@@ -100,6 +100,44 @@ impl GeoFeatureCollectionRowBuilder<MultiPoint> for FeatureCollectionRowBuilder<
     }
 }
 
+impl MultiPointCollection {
+    pub fn coordinates(&self) -> &[Coordinate2D] {
+        let geometries_ref = self
+            .table
+            .column_by_name(Self::GEOMETRY_COLUMN_NAME)
+            .expect("There must exist a geometry column");
+        let geometries: &ListArray = downcast_array(geometries_ref);
+
+        let coordinates_ref = geometries.values();
+        let coordinates: &FixedSizeListArray = downcast_array(&coordinates_ref);
+
+        let number_of_coordinates = coordinates.data().len();
+
+        let floats_ref = coordinates.values();
+        let floats: &Float64Array = downcast_array(&floats_ref);
+
+        unsafe {
+            slice::from_raw_parts(
+                floats.raw_values() as *const Coordinate2D,
+                number_of_coordinates,
+            )
+        }
+    }
+
+    pub fn coordinates_offsets(&self) -> &[i32] {
+        let geometries_ref = self
+            .table
+            .column_by_name(Self::GEOMETRY_COLUMN_NAME)
+            .expect("There must exist a geometry column");
+        let geometries: &ListArray = downcast_array(geometries_ref);
+
+        let data = geometries.data();
+        let buffer = &data.buffers()[0];
```

Comment:
You didn't check for an offset within the buffer.
You would need to test that somehow by subsetting the geometry column.

Reply:
when is the offset not zero? I need a scenario to test it

Reply:
I guess when a buffer is a subset of another one.

Reply:
can that happen here though?

Reply:
I guess that depends if we ever will use the `slice` functionality of arrow to slice a collection.

Reply:
Currently there are two ways to build a feature collection: RowBuilder and BatchBuilder. In both cases the buffers are not built by slicing. Maybe you can design a test case that fails for the current implementation. Or: we leave it for now.